### PR TITLE
Sf item

### PR DESCRIPTION
**Pull request by s-yfang**
I just did this assuming that we'll be using a txt file until we think of something better

### CODE REVIEW

File: Item.cpp [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	std::string* stats = new std::string[5]{};
+
+	int i = 0;
+
+	while (ssline.good() && i < 5) {
+		ssline >> stats[i];
+		++i;
+	}
+
+	type = stats[0];								// Specifies what class of item it is
```
Comment (zzakiullah) [inline Item.cpp:20]: Does it work without doing Item::type? Same for the others below

File: Item.cpp [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	value = std::stoi(stats[4]);					// Value of the item determines its price
+
+	price = value + rarity * durability;			// An example of how we could calculate it
+
+}
+
+Item::~Item() {}
+
+// Getters
+std::string Item::get_type() { return type; }
```
Comment (zzakiullah) [inline Item.cpp:33]: Same question as before (does it work without doing Item::type?)

File: Item.cpp [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+int Item::get_itemID() { return itemID; }
+int Item::get_rarity() { return rarity; }
+double Item::get_durability() { return durability; }
+double Item::get_value() { return value; }
+double Item::get_price() { return price; }
+
+// Setters
+void Item::set_type(std::string t) { type = t; }
```
Comment (zzakiullah) [inline Item.cpp:41]: Same question as before (does it work without doing Item::type = t?)

**Review by zzakiullah — CHANGES_REQUESTED** [review]

File: Item.cpp (hunk 1 above)
Comment (s-yfang) [reply Item.cpp:20]: Do you mean without Item::type = stats[0];? I think so, it doesn't show any errors.

Comment (zzakiullah) [reply Item.cpp:20]: Oh lol, didn't know that. I'll have to go back and change the other classes then since this is shorter

File: Item.cpp [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+	int i = 0;
+
+	while (ssline.good() && i < 5) {
+		ssline >> stats[i];
+		++i;
+	}
+
+	type = stats[0];								// Specifies what class of item it is
+	itemID = std::stoi(stats[1]);					// Item name/more specific 
+	durability = std::stoi(stats[2]);
```
Comment (zzakiullah) [inline Item.cpp]: durability is a double so I think you should use std::stod instead of std::stoi

File: Item.cpp [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+	rarity = std::stoi(stats[3]);
+	value = std::stoi(stats[4]);					// Value of the item determines its price
```
Comment (zzakiullah) [inline Item.cpp]: value is a double so I think you should use std::stod instead of std::stoi

**Review by zzakiullah — CHANGES_REQUESTED** [review]

File: Item.cpp (hunk 5 above)
Comment (s-yfang) [reply Item.cpp]: Oh yeah thanks, missed that

**Review by rabiyamajeed — APPROVED** [review]
epic

**Review by zzakiullah — APPROVED** [review]
Coolio

**Pull request merged**